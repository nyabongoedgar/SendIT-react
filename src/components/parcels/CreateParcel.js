import React from 'react'

const CreateParcel = (props) => {
    let Loader = require('react-loader')
    return (
        <div className="authForms">
       
        <form onSubmit={props.submitHandler} className="authForms">
        <h1>CREATE PARCEL DELIVERY ORDER</h1>
        <input onChange={props.inputHandler} value={props.parcel_description} type="text" placeholder="parcel_description" name="parcel_description" required />
                                
        <input onChange={props.inputHandler} value={props.parcel_weight} type="number" name="parcel_weight" placeholder="parcel_weight" required/> <br/> 
                        
        <input onChange={props.inputHandler} value={props.parcel_source} type="text" name="parcel_source" placeholder="parcel_source" required/><br/>

        <input onChange={props.inputHandler} value={props.parcel_destination} type="text" name="parcel_destination" placeholder=" parcel_destination" required /> <br/>

        <input onChange={props.inputHandler} value={props.receiver_name} type="text" name="receiver_name" placeholder="Enter receiver name" required />
        <br/>
        <input onChange={props.inputHandler} value={props.receiver_telephone} maxLength="10" type="text" name="receiver_telephone" placeholder="Enter receiver Telephone number starting with a 0" required />
        <Loader loaded={!props.isLoading}>
        <button type="submit">CREATE ORDER</button>
        </Loader>
        </form>
        
        </div>
    );
};

export default CreateParcel;