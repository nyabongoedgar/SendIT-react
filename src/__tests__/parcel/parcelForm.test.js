import React from 'react';
import CreateParcel from '../../components/parcels/CreateParcel';
import renderer from 'react-test-renderer';


describe('SignupForm Component', () => {

  it('it matches the snapshot', () => {
    const tree = renderer.create(<CreateParcel />).toJSON();
    expect(tree).toMatchSnapshot();
  });


});
