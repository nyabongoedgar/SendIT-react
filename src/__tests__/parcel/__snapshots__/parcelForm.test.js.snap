// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupForm Component it matches the snapshot 1`] = `
<div
  className="authForms"
>
  <form
    className="authForms"
  >
    <h1>
      CREATE PARCEL DELIVERY ORDER
    </h1>
    <input
      name="parcel_description"
      placeholder="parcel_description"
      required={true}
      type="text"
    />
    <input
      name="parcel_weight"
      placeholder="parcel_weight"
      required={true}
      type="number"
    />
     
    <br />
    <input
      name="parcel_source"
      placeholder="parcel_source"
      required={true}
      type="text"
    />
    <br />
    <input
      name="parcel_destination"
      placeholder=" parcel_destination"
      required={true}
      type="text"
    />
     
    <br />
    <input
      name="receiver_name"
      placeholder="Enter receiver name"
      required={true}
      type="text"
    />
    <br />
    <input
      maxLength="10"
      name="receiver_telephone"
      placeholder="Enter receiver Telephone number starting with a 0"
      required={true}
      type="text"
    />
    <div
      className="loadedContent"
    >
      <button
        className="btn"
        type="submit"
      >
        CREATE ORDER
      </button>
    </div>
  </form>
</div>
`;
